### DISCUSSION TITLE
It seems that ”ccache“ doesn't work with CMAKE_TOOLCHAIN_FILE “emscripten.cmake”

### DISCUSSION THREAD

**Discussion by DOOOBETTER** [ccache/ccache#1545] [Questions & answers]
Hello all,
 when I was trying to add "ccache" to my cmake program,  no new files were added to my ccache directory. But I tried test "ccache em++/emcc" in a small project, it does work.

 `cmake -B build_wasm -S . -DCMAKE_TOOLCHAIN_FILE=/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_BUILD_TYPE=MinSizeRel -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache  ` 
I use this way to apply ccache to my cmake file.

**Comment by DOOOBETTER** [comment]
~`I tried several ways to fix my problem and found that adding " export CCACHE_COMPILER_TYPE=emcc" to my script file can work out`~

**Reply by jrosdahl** [reply]
Great that you solved your problem!

However, `export CCACHE_COMPILER_TYPE=emcc` doesn't make sense since ccache doesn't know about such a compiler type: https://ccache.dev/manual/4.10.2.html#config_compiler_type

**Reply by DOOOBETTER** [reply]
sorry, I made a mistake. The real reason why my "ccache" doesn't work is that I didn't use an absolute path to set the "CCACHE_DIR".
